web/src/components/storage/VolumeForm.jsx

```diff
@@ -0,0 +1,478 @@
+          <NumericTextInput
+            id="size"
+            name="size"
+            aria-label="Desired size"
+            value={formData.size}
+            onChange={(size) => onChange({ size })}
+            validated={errors.size && 'error'}
+          />
+          <SizeUnitFormSelect
+            id="sizeUnit"
+            aria-label="Size unit"
+            units={Object.values(SIZE_UNITS)}
+            value={formData.sizeUnit || DEFAULT_SIZE_UNIT}
+            onChange={(sizeUnit) => onChange({ sizeUnit })}
+          />
+        </InputGroup>
+      </FormGroup>
+    </div>
+  );
+};
+
+/**
+ * Widget for rendering the size option content when SIZE_UNITS.RANGE is selected
+ * @component
+ *
+ * @param {object} props
+ * @param {object} props.errors - the form errors
+ * @param {object} props.formData - the form data
+ * @param {onChangeFn} props.onChange - callback for notifying input changes
+ *
+ * @returns {ReactComponent}
+ */
+const SizeRange = ({ errors, formData, onChange }) => {
+  return (
+    <div className="stack">
+      <p>
+        Limits for the file system size. The final size will be a value between the given minimum
+        and maximum sizes. If no maximum is given, then the file system will be as big as possible.
+      </p>
+      <div className="split" data-items-alignment="start">
+        <FormGroup
+          isRequired
+          label="Minimum"
+          fieldId="minSize"
+          className="size-input-group"
+          validated={errors.minSize && 'error'}
+          helperTextInvalid={errors.minSize}
+        >
+          <InputGroup>
+            <NumericTextInput
+              id="minSize"
+              name="minSize"
+              aria-label="Minimum desired size"
+              value={formData.minSize}
+              onChange={(minSize) => onChange({ minSize })}
+              validated={errors.minSize && 'error'}
+            />
+            <SizeUnitFormSelect
+              id="minSizeUnit"
+              aria-label="Min size unit"
+              units={Object.values(SIZE_UNITS)}
+              value={formData.minSizeUnit || DEFAULT_SIZE_UNIT }
+              onChange={(minSizeUnit) => onChange({ minSizeUnit })}
+            />
+          </InputGroup>
+        </FormGroup>
+        <FormGroup
+          label="Maximum"
+          fieldId="maxSize"
+          className="size-input-group"
+          validated={errors.maxSize && 'error'}
+          helperTextInvalid={errors.maxSize}
+        >
+          <InputGroup>
+            <NumericTextInput
+              id="maxSize"
+              name="maxSize"
+              validated={errors.maxSize && 'error'}
+              aria-label="Maximum desired size"
+              value={formData.maxSize}
+              onChange={(maxSize) => onChange({ maxSize })}
+
+            />
+            <SizeUnitFormSelect
+              id="maxSizeUnit"
+              aria-label="Max size unit"
+              units={Object.values(SIZE_UNITS)}
```

Comment:
Perhaps it can be part of the `SizeUnitFormSelect`. After all, we are always giving the same value.

Reply:
And initially it was part of it. But then I extracted it because _a premature improvement_, thinking in a context where the selector cuold be used with a narrowed set of units. Although it's true that

- the SizeUnitFormSelect is not exposed yet
- is is using always the same units in this form

If you don't mind, I'm going to keep it as it is now.